fondant/component_spec.py

```diff
@@ -300,58 +147,146 @@ def output_subsets(self) -> t.Mapping[str, ComponentSubset]:
+            )
+            for arg_name, arg_info in self._specification["args"].items()
+        ]
+
+    @property
+    def kubeflow_specification(self) -> "KubeflowComponentSpec":
+        return KubeflowComponentSpec.from_fondant_component_spec(self)
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self._specification!r}"
+
+
+class KubeflowComponentSpec:
+    """
+    Class representing a Kubeflow component specification.
+
+    Args:
+        specification: The component specification as a Python dict
+    """
+
+    def __init__(self, specification: t.Optional[dict]) -> None:
+        self._specification = specification
+
+    @classmethod
+    def from_fondant_component_spec(
+        cls, fondant_component: FondantComponentSpec
+    ) -> "KubeflowComponentSpec":
+        """Create a Kubeflow component spec from a Fondant component spec."""
+        specification = {
+            "name": fondant_component.name,
+            "description": fondant_component.description,
+            "inputs": [
+                {
+                    "name": "input_manifest_path",
+                    "description": "Path to the input manifest",
+                    "type": "String",
+                },
+                *(
+                    {
+                        "name": arg.name,
+                        "description": arg.description,
+                        "type": python2kubeflow_type[arg.type],
+                    }
+                    for arg in fondant_component.args
+                ),
+            ],
+            "outputs": [
+                {
+                    "name": "output_manifest_path",
+                    "description": "Path to the output manifest",
+                    "type": "String",
+                },
+            ],
+            "implementation": {
+                "container": {
+                    "image": fondant_component.image,
+                    "command": [
+                        "python3",
+                        "main.py",
+                        "--input-manifest-path",
+                        {"inputPath": "input_manifest_path"},
+                        *cls._dump_args(fondant_component.args),
+                        "--output-manifest-path",
+                        {"outputPath": "output_manifest_path"},
+                    ],
+                }
+            },
+        }
+        return cls(specification)
+
+    @staticmethod
+    def _dump_args(args: t.List[Argument]) -> t.List[t.Union[str, t.Dict[str, str]]]:
+        """Dump Fondant specification arguments to KfP command arguments."""
+        dumped_args = []
+        for arg in args:
+            arg_name = arg.name.replace("-", "_").strip()
+            arg_name_cmd = f'--{arg.name.replace("_", "-")}'.strip()
+
+            dumped_args.append(arg_name_cmd)
+            dumped_args.append({"inputValue": arg_name})
+
+        return dumped_args
+
+    def to_file(self, path: t.Union[str, Path]) -> None:
```

Comment:
I noticed that when compiling the kfp pipeline you can pass in directly a json string instead of the yaml file. So might be worth switching over to that approach. But this should be handled in separate PR related when automating the pipeline definition and compilation

Reply:
Ok, the json is available under `_spec` already, so should be easy to expose.